plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAlluxioCacheFileOperationsMutableTransactionLog.java

```diff
@@ -0,0 +1,463 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMultiset;
+import com.google.common.io.Resources;
+import io.trino.filesystem.tracing.CacheFileSystemTraceUtils;
+import io.trino.testing.DistributedQueryRunner;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.stream.Stream;
+
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static io.trino.plugin.deltalake.TestingDeltaLakeUtils.copyDirectoryContents;
+import static java.lang.String.format;
+
+@Execution(ExecutionMode.SAME_THREAD)
+public class TestDeltaLakeAlluxioCacheFileOperationsMutableTransactionLog
+        extends TestDeltaLakeAlluxioCacheFileOperations
+{
+    @Override
+    protected DistributedQueryRunner createQueryRunner()
+            throws Exception
+    {
+        Path cacheDirectory = Files.createTempDirectory("cache");
+        closeAfterClass(() -> deleteRecursively(cacheDirectory, ALLOW_INSECURE));
+
+        return DeltaLakeQueryRunner.builder()
+                .setCoordinatorProperties(ImmutableMap.of("node-scheduler.include-coordinator", "false"))
+                .setDeltaProperties(ImmutableMap.<String, String>builder()
+                        .put("fs.cache.enabled", "true")
+                        .put("fs.cache.directories", cacheDirectory.toAbsolutePath().toString())
+                        .put("fs.cache.max-sizes", "100MB")
+                        .put("delta.enable-non-concurrent-writes", "true")
+                        .put("delta.register-table-procedure.enabled", "true")
+                        .put("delta.fs.cache.disable-transaction-log-caching", "true")
+                        .buildOrThrow())
+                .setWorkerCount(1)
+                .build();
+    }
+
+    @Override
+    @Test
+    public void testCacheFileOperations()
```

Comment:
My reasoning was that, since deltaLogFileSystemCacheDisabled can have two different values, I would need to test the equivalent of each scenario with the delta log disabled. If its an overkill I will create a minimal test case as suggested.

Reply:
We don't need all the tests here, a minimal test case which covers the new functionality is sufficient.

Reply:
Thanks for the feedback, I will write a minimal test.

Reply:
Since I needed `assertFileSystemAccesses` for my test, I created an intermediate abstract class and moved it there. 
Please let me know if you see any issues with this.